core/src/channel/mod.rs

```diff
@@ -249,6 +267,37 @@ impl VoiceChannelData {
                         self.process_control_event(ControlEvent::PitchBendSensitivity(sensitivity))
                     }
                 }
+                0x07 => {
+                    // Volume
+                    let vol: f32 = value as f32 / 128.0;
+                    self.control_event_data
+                    .borrow_mut()
+                    .volume = vol
+                }
+                0x0A => {
+                    // Pan
+                    let pan: f32 = value as f32 / 128.0;
+                    self.control_event_data
+                    .borrow_mut()
+                    .pan = pan
+                }
+                0x40 => {
+                    // Damper / Sustain
+                    let damper = match value {
+                        0..=63 => false,
+                        64..=127 => true,
+                        _ => false,
+                    };
+
+                    for key in self.key_voices.iter() {
+                        key.data.borrow().voices.damper = damper;
```

Comment:
As I mentioned in another comment, it would be better to use a set_damper function here.